index.tsx

```diff
@@ -0,0 +1,353 @@
+    withAnimation,
+  ]);
+
+  const onDoubleTap = useCallback((): void => {
+    if (isZoomedIn.value) {
+      zoomOut();
+    } else {
+      zoomIn();
+    }
+  }, [zoomIn, zoomOut, isZoomedIn]);
+
+  const onLayout = useCallback(({nativeEvent: {layout: {width, height}}}: LayoutChangeEvent): void => {
+    containerDimensions.value = {
+      width,
+      height,
+    };
+  }, [containerDimensions]);
+
+  const onLayoutContent = useCallback(({nativeEvent: {layout: {width, height}}}: LayoutChangeEvent): void => {
+    contentDimensions.value = {
+      width,
+      height,
+    };
+  }, [contentDimensions]);
+
+  const onPinchEnd = useCallback((scale: number): void => {
+    const newScale = lastScale.value * scale;
+    lastScale.value = newScale;
+    if (newScale > 1) {
+      isZoomedIn.value = true;
+      baseScale.value = newScale;
+      pinchScale.value = 1;
+
+      handlePanOutside();
+    } else {
+      zoomOut();
+    }
+  }, [
+    lastScale,
+    baseScale,
+    pinchScale,
+    handlePanOutside,
+    zoomOut,
+    isZoomedIn,
+  ]);
+
+  const panOffsetsBeforeGestureStart: SharedValue<{ x: number | null; y: number | null }> = useSharedValue({
+    x: null,
+    y: null,
+  });
+
+  const zoomGestures = useMemo(() => {
+    const tapGesture = Gesture.Tap()
+      .numberOfTaps(2)
+      .onEnd(() => {
+        runOnJS(onDoubleTap)();
+      });
+
+    const panGesture = Gesture.Pan()
+      .onUpdate((event: GestureUpdateEvent<PanGestureHandlerEventPayload>): void => {
+        let {translationX, translationY} = event;
+
+        if (panOffsetsBeforeGestureStart.value.x === null || panOffsetsBeforeGestureStart.value.y === null) {
+          panOffsetsBeforeGestureStart.value.x = translationX;
+          panOffsetsBeforeGestureStart.value.y = translationY;
+        }
+
+        translationX -= panOffsetsBeforeGestureStart.value.x;
+        translationY -= panOffsetsBeforeGestureStart.value.y;
+
+        translateX.value = lastOffsetX.value + translationX / lastScale.value;
+        translateY.value = lastOffsetY.value + translationY / lastScale.value;
+      })
+      .onEnd((event: GestureStateChangeEvent<PanGestureHandlerEventPayload>): void => {
+        let {translationX, translationY} = event;
+
+        if (panOffsetsBeforeGestureStart.value.x !== null && panOffsetsBeforeGestureStart.value.y !== null) {
+          translationX -= panOffsetsBeforeGestureStart.value.x;
+          translationY -= panOffsetsBeforeGestureStart.value.y;
+        }
+
+        lastOffsetX.value = lastOffsetX.value + translationX / lastScale.value;
+        lastOffsetY.value = lastOffsetY.value + translationY / lastScale.value;
+
+        panOffsetsBeforeGestureStart.value.x = null;
+        panOffsetsBeforeGestureStart.value.y = null;
+
+        runOnJS(handlePanOutside)();
+      })
+      .onTouchesMove((e: GestureTouchEvent, state: GestureStateManagerType): void => {
+        if ([State.UNDETERMINED, State.BEGAN].some(s => s === e.state)) {
```

Comment:
Doesn't look like it's more readable than "includes"

Reply:
reverted

Reply:
It's because of the TS compiler I configured. it does not know `includes` function. if you could fix it, then that's ok to revert it.

Reply:
@kesha-antonov 

I change it because of this error. if you could fix it using another approach, then that's ok to revert it.

<img width="1229" alt="image" src="https://github.com/kesha-antonov/react-native-zoom-reanimated/assets/13766432/10d0c3c5-6a32-469f-9993-16ae2e9cf0c0">